numpy/core/src/umath/_umath_tests.c.src

```diff
@@ -195,6 +197,41 @@ static void
 
 /**end repeat**/
 
+char *cross1d_signature = "(3),(3)->(3)";
+
+/**begin repeat
+
+   #TYPE=LONG,DOUBLE#
+   #typ=npy_long, npy_double#
+*/
+
+/*
+ *  This implements the cross product:
+ *        out[n, 0] = in1[n, 1]*in2[n, 2] - in1[n, 2]*in2[n, 1]
+ *        out[n, 1] = in1[n, 2]*in2[n, 0] - in1[n, 0]*in2[n, 2]
+ *        out[n, 2] = in1[n, 0]*in2[n, 1] - in1[n, 1]*in2[n, 0]
+ */
+static void
+@TYPE@_cross1d(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
+{
+    INIT_OUTER_LOOP_3
+    npy_intp is1=steps[0], is2=steps[1], os = steps[2];
+    BEGIN_OUTER_LOOP_3
+        char *ip1=args[0], *ip2=args[1], *op=args[2];
```

Comment:
Better I think to introduce some locals
```C
@typ@ i1_x = *(@typ@ *)(args[0] + 0*is1);
@typ@ i1_y = *(@typ@ *)(args[0] + 1*is1);
@typ@ i1_z = *(@typ@ *)(args[0] + 2*is1);

@typ@ i2_x = *(@typ@ *)(args[1] + 0*is2);
@typ@ i2_y = *(@typ@ *)(args[1] + 1*is2);
@typ@ i2_z = *(@typ@ *)(args[1] + 2*is2);
```
Which avoids the mix of dereferencing and multiplication below.

Reply:
adopted